### PR TITLE
[ZEPPELIN-5857] Bump gRPC and protobuf to support build on macOS M1 chips

### PR DESCRIPTION
**Pull request by pan3793**
### What is this PR for?

I suppose this upgrade will not introduce compatibility issue, see details discussion about version requirements and compatibility https://github.com/apache/incubator-uniffle/pull/13

### What type of PR is it?

Improvement

### Todos
* [ ] - Task

### What is the Jira issue?

ZEPPELIN-5857

### How should this be tested?

Build on my M1 mac.

### Screenshots (if appropriate)

before
image

after
image

### Questions:
* Does the licenses files need to update? Yes, and updated.
* Is there breaking changes for older versions? No.
* Does this needs documentation? No.

### CODE REVIEW

**Comment by jongyoul** [comment]
I've checked the link you added but it's a bit different because we are using another maven plugin so I cannot reproduce the error like you. Could you please check it? BTW, I like the approach you suggested but we'd better change more to make them the same. Could you please check and modify them?

**Comment by pan3793** [comment]
@jongyoul are you using x86 JDK w/ Rosetta? the native aarch64 JDK should fail.
```
➜  apache-zeppelin git:(ZEPPELIN-5857) java -version
openjdk version "1.8.0_332"
OpenJDK Runtime Environment (Zulu 8.62.0.19-CA-macos-aarch64) (build 1.8.0_332-b09)
OpenJDK 64-Bit Server VM (Zulu 8.62.0.19-CA-macos-aarch64) (build 25.332-b09, mixed mode)
```

**Comment by pan3793** [comment]
typically, the native aarch64 JDK is 2x faster than x86 JDK w/ Rosetta (testing by compiling Apache Spark and Apache Kyuubi)

**Comment by jongyoul** [comment]
I tested it failed with aarch64 jdk. Thank you for letting me know.

File: zeppelin-jupyter-interpreter-shaded/pom.xml [hunk 1]
```diff
@@ -36,6 +36,28 @@
...
+    <dependency>
+      <groupId>com.google.protobuf</groupId>
+      <artifactId>protobuf-java</artifactId>
+      <version>${protobuf.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.errorprone</groupId>
+      <artifactId>error_prone_annotations</artifactId>
+      <version>2.14.0</version>
```
Comment (jongyoul) [inline zeppelin-jupyter-interpreter-shaded/pom.xml:60]: Could you please follow our current format to define the version of this dependency?

Comment (pan3793) [reply zeppelin-jupyter-interpreter-shaded/pom.xml]: done

**Comment by pan3793** [comment]
@jongyoul this is ready for review, I believe the CI failure is irrelevant.

I can success build the project(except to frontend modules) using Java8 aarch64 on M1 mac now.

**Review by jongyoul — APPROVED** [review]

**Pull request merged**